src/main/java/reposense/report/ReportGenerator.java

```diff
@@ -93,23 +93,18 @@
         earliestSinceDate = null;
         progressTracker = new ProgressTracker(configs.size());
 
-        cloneAndAnalyzeRepos(configs, outputPath);
+        List<Path> reportFoldersAndFiles = cloneAndAnalyzeRepos(configs, outputPath);
 
         Date reportSinceDate = (cliSinceDate.equals(SinceDateArgumentType.ARBITRARY_FIRST_COMMIT_DATE))
                 ? earliestSinceDate : cliSinceDate;
 
         FileUtil.writeJsonFile(
                 new SummaryJson(configs, generationDate, reportSinceDate, untilDate, isSinceDateProvided,
                         isUntilDateProvided, RepoSense.getVersion(), ErrorSummary.getInstance().getErrorList()),
-                getSummaryResultPath(outputPath));
+                getSummaryResultPath(outputPath))
+            .ifPresent(reportFoldersAndFiles::add);
```

Comment:
This is starting to get unreadable.
Instead of this, can we split getting and adding the path name into two?

```java
Optional<Path> summaryPath = FileUtil.writeJsonFile(...);
summaryPath.ifPresent(reportFoldersAndFiles::add);
```

Reply:
Thanks for the suggestion. I've changed it.